internal/component/azure/servicebus/subscription.go

```diff
@@ -361,8 +361,8 @@ func (s *Subscription) RenewLocksBlocking(ctx context.Context, receiver Receiver
 				s.logger.Debugf("Renewed session %s locks for %s", sessionReceiver.SessionID(), s.entity)
 			} else {
 				// Snapshot the messages to try to renew locks for.
-				msgs := make([]*azservicebus.ReceivedMessage, len(s.activeMessages))
 				s.mu.RLock()
+				msgs := make([]*azservicebus.ReceivedMessage, len(s.activeMessages))
 				for i, m := range s.activeMessages {
 					msgs[i] = m
```

Comment:
@jjcollinge Should this be `msgs = append(msgs, m)`?
Since activeMessages is a map and key is a sequenceNumber which can be more than the index value of the list `msgs`.

Reply:
Ah yes, good spot - I will revert to append 👌

Reply:
No need for append then, which is not efficient with allocs 

use a separate I counter variable